Adding code for Longest Substring Without Repeating Characters

anilhariparamhans opened the pull request:
https://leetcode.com/problems/longest-substring-without-repeating-characters/

Commit:
Adding code for

Longest Substring Without Repeating Characters

https://leetcode.com/problems/longest-substring-without-repeating-characters/

dimplenaresh1 commented on LeetCode/vAnil/Practice/LongestSubstrWithoutRepeatingChars/LongestSubstrWithoutRepeatingChars.cpp:
```diff
@@ -0,0 +1,30 @@
+		}
+	}
+
+	return maxSeqLen;
```
Greater of maxSeqLen and CurrSeqCount should be returned.

anilhariparamhans replied:
Good catch !

I incorporated this review comment.

Commit:
Adding code for
Reverse Words in a String

https://leetcode.com/problems/reverse-words-in-a-string/

Commit:
Incorporated review comments

Commit:
Merge branch 'master' of https://github.com/DimpleOrg/InterviewDS into HeapBranch

dimplenaresh1 commented on LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:
```diff
@@ -0,0 +1,86 @@
+		}
+	}
+
+	return str;
```
getting SEGV error when submitting in Leetcode.

anilhariparamhans replied:
Fixed this.

dimplenaresh1 commented on LeetCode/vAnil/Practice/PracticeQuestions/ReverseWordInString.cpp:
```diff
@@ -0,0 +1,86 @@
+		else
+			break;
+	}
+	str.erase(0, count);
```
out of range subscript error if str is ""

anilhariparamhans replied:
Fixed it.

Commit:
Incoprotated the review comments.

And submitted on LeetCode. Got this result

Runtime: 8 ms, faster than 99.68% of C++ online submissions for Reverse Words in a String.
Memory Usage: 9.6 MB, less than 41.55% of C++ online submissions for Reverse Words in a String.

anilhariparamhans commented:
@dimplenaresh1 fixed the issues
and got this result on Leetcode

Runtime: 8 ms, faster than 99.68% of C++ online submissions for Reverse Words in a String.
Memory Usage: 9.6 MB, less than 41.55% of C++ online submissions for Reverse Words in a String.

Commit:
Merge branch 'master' of https://github.com/DimpleOrg/InterviewDS into HeapBranch
